lxd/main_forkmount.go

```diff
@@ -152,14 +153,35 @@ void do_lxd_forkmount(pid_t pid) {
 		_exit(1);
 	}
 
+	if (strcmp(shiftfs, "true") == 0) {
+		// Setup shiftfs inside the container
+		if (mount(src, src, "shiftfs", 0, "passthrough=3") < 0) {
+			fprintf(stderr, "Failed shiftfs setup for %s: %s\n", src, strerror(errno));
+			_exit(1);
+		}
+	}
+
 	// Here, we always move recursively, because we sometimes allow
 	// recursive mounts. If the mount has no kids then it doesn't matter,
 	// but if it does, we want to move those too.
 	if (mount(src, dest, "none", MS_MOVE | MS_REC, NULL) < 0) {
+		// If using shiftfs, undo the shiftfs mount
+		if (strcmp(shiftfs, "true") == 0) {
+			umount2(src, MNT_DETACH);
+		}
```

Comment:
We don't use `{}` around single-line if statements. :) But again, no blocker. :)